lib/Driver/ToolChains.cpp

```diff
@@ -1270,8 +1293,37 @@ toolchains::GenericUnix::constructInvocation(const LinkJobAction &job,
     Arguments.push_back(context.Args.MakeArgString(context.OI.SDKPath));
   }
 
+  // Link the standard library.
   Arguments.push_back("-L");
-  Arguments.push_back(context.Args.MakeArgString(RuntimeLibPath));
+  if (context.Args.hasFlag(options::OPT_static_stdlib,
+                            options::OPT_no_static_stdlib,
+                            false)) {
+    SmallString<128> StaticRuntimeLibPath;
+    getRuntimeStaticLibraryPath(StaticRuntimeLibPath, context.Args, *this);
+    Arguments.push_back(context.Args.MakeArgString(StaticRuntimeLibPath));
+    // The following libraries are required to build a satisfactory
+    // static program
+    Arguments.push_back("-ldl");
+    Arguments.push_back("-lpthread");
+    Arguments.push_back("-lbsd");
+    Arguments.push_back("-licui18n");
+    Arguments.push_back("-licuuc");
+    // The runtime uses dlopen to look for the protocol conformances.
+    // Therefore, we need to ensure they appear in the dynamic table.
+    // This happens automatically for dynamically-linked programs, but
+    // in this case we have to take additional measures.
+    const char *dynamicList = context.getTemporaryFilePath("export-protocol-conformances", "");
+    const char dynamicListContents[] = "{\n.swift2_protocol_conformances_start;\n};";
+    FILE *dynamicListFile = fopen (dynamicList,"w");
+    assert(dynamicListFile);
+    int result = fputs(dynamicListContents, dynamicListFile);
+    assert(result != EOF);
+    fclose(dynamicListFile);
```

Comment:
@jrose-apple You know if there's an LLVM utility API we could use for writing a string to a temporary file here, instead of using bare stdio?